README.md

```diff
@@ -785,3 +788,31 @@ The React Native community has graciously created some awesome open source apps
 * [MoveIt!](https://github.com/multunus/moveit-react-native) - An app by [Multunus](http://www.multunus.com) that allows employees within a company to track their work-outs.
 
 *Note: If you've developed a React Native app using CodePush, that is also open-source, please let us know. We would love to add it to this list!*
+
+## Debugging / Troubleshooting
+
+The `sync` method includes a lot of diagnostic logging out-of-the-box, so if you're encountering an issue when using it, the best thing to try first is examining the output logs of your app. This will tell you whether the app is configured correctly (e.g. can the plugin find your deployment key?), if the app is able to reach the server, if an available update is being discovered, if the update is being successfully downloaded/installed, etc. We want to continue improving the logging to be as intutuive/comprehensive as possible, so please [let us know](mailto:codepushfeed@microsoft.com) if you find it to be confusing or missing anything.
+
+![Xcode Console](https://cloud.githubusercontent.com/assets/116461/13536459/d2687bea-e1f4-11e5-9998-b048ca8d201e.png)
+
+To view these logs, you can use either the Chrome DevTools Console, the XCode Console (iOS) and/or ADB logcat (Android). By default, React Native logs are disabled on iOS in release builds, so if you want to view them in a release build, you simply need to make the following changes to your `AppDelegate.m` file:
+
+1. Add an `#import "RCTLog.h"` statement
+
+2. Add the following statement to the top of your `application:didFinishLaunchingWithOptions` method:
+
+    ```objective-c
+    RCTSetLogThreshold(RCTLogLevelInfo);
+    ```
+
+Now you'll be able to see CodePush logs in either debug or release mode, on both iOS or Android. If examining the logs don't provide an indication of the issue, please refer to the following common issues for additional resolution ideas:
+
+| Issue / Symptom | Possible Solution |
+|-----------------|-------------------|
+| Compilation Error | Double check that your version of React Native is [compatible](#supported-react-native-platforms) with the CodePush version you are using |
+| Network timeout / hang when calling `sync` or `checkForUpadte` in the iOS Simulator | Try resetting it the simulator by selecting the `Simulator -> Reset Content and Settings..` menu item. |
+| Server responds with a `404` when calling `sync` or `checkForUpdate` | Double-check that the deployment key you added to your `Info.plist` (iOS), `build.gradle` (Android) or that you're passing to `sync`/`checkForUpdate` is correct. You can run `code-push deployment ls [APP_NAME] -k` to view the correct keys for your app deployments. |
+| Update not being discovered | Double-check that the version of your running app (e.g. `1.0.0`) matches the version you specified when releasing the update to CodePush. Additionally, make sure that you are releasing to the same deployment that you're app is configured to sync with. |
```

Comment:
Additionally, make sure that you are releasing to the same deployment that **your** app is configured to sync with.